Mathlib/Algebra/Algebra/Basic.lean

```diff
@@ -285,16 +285,16 @@ section FaithfulSMul
 
 instance (R : Type*) [NonAssocSemiring R] : FaithfulSMul R R := ⟨fun {r₁ r₂} h ↦ by simpa using h 1⟩
 
-variable (R A : Type*) [CommSemiring R] [Semiring A]
+variable (R A : Type*) [Semiring A]
 
-lemma faithfulSMul_iff_injective_smul_one [Module R A] [IsScalarTower R A A] :
+lemma faithfulSMul_iff_injective_smul_one [SMul R A] [IsScalarTower R A A] :
```

Comment:
In fact,

- https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Basic.html#instFaithfulSMul can be generalized to `MulOneClass` (and moved way earlier in the `import` graph), replacing `RightCancelMonoid.faithfulSMul`;
- this lemma can be generalized to this (I assume `[mk_iff]` on `FaithfulSMul`):
```lean
example {M N α : Type*} [SMul M N] [Monoid N] [MulAction N α] [SMul M α]
    [IsScalarTower M N α] [FaithfulSMul N α] :
    FaithfulSMul M α ↔ Function.Injective (· • 1 : M → N) := by
  simp_rw [faithfulSMul_iff, ← (smul_left_injective' (M := N) (α := α)).of_comp_iff,
    Function.comp_def, smul_one_smul, Function.Injective, funext_iff]
```
(probably, under a different name), then specialized to the case `N = α` under the current name.

Reply:
I was unimaginative and came up with the primed name for the new lemma, please let me know if there's some other convention I should use in cases like this.